### ISSUE TITLE
GoogleAppScriptの定期実行について

### ISSUE THREAD

**Issue by kiripeace** [NIFCloud-mbaas/UserCommunity#833]
いつもお世話になっております。

スクリプトの定期実行を行いたく、GoogleAppScriptを使用しています。

niftyのサイトからスクリプトを選択し実行する分にはエラーが発生しないのですが
GoogleAppScriptにてタイマーを設定し実行しようとすると

「SyntaxError: Unexpected token: U（行 84、ファイル「Code」、プロジェクト「NCMB」）」

といったようなエラーが毎回発生してしまいます。

なにか原因の分かる方いましたらご教授お願い致します。

GoogleAppScriptのコード
`function pushFlagDelete() {`
`  var application_key = "自分のアプリケーションキー";`
  `var client_key = "自分のクライアントキー";`
  `var ncmb = NCMB.init(application_key, client_key);`
 ` var result = ncmb.Script()`
                 `  .exec("GET", "pushFlagDelete.js");`
`}`

**Comment by goofmint** [comment]
Google Apps ScriptはJavaScriptのサブセットで、通常のJavaScript SDKは動きません。

私が作ったGoogle Apps Script向けのNCMBライブラリがあるのでお試しください。

http://blog.mbaas.nifcloud.com/entry/11125

**Comment by kiripeace** [comment]
@goofmint 
申し訳ありません。説明が足りていませんでした。
http://blog.mbaas.nifcloud.com/entry/11125
コードは貼っていただいたこちらのリンクを参考にし、ライブラリをいれて作成を行っています。

エラーは発生するもののスクリプトの実行自体は行われていて、
エラー多発のため実行に制限がかけられている状態になります。。。

動いてはいるのですが、エラー多発で制限をかけられてしまうと困るので
今回エラーを消す方法を探しているという状況です。

**Comment by goofmint** [comment]
Google Apps Scriptの内容を見るに、JSONでパースした結果を返そうとしています。それが失敗しているようです（返り値がJSONテキストではないのかも？）。mBaaS側のスクリプトのログはどうなっているでしょうか？

**Comment by kiripeace** [comment]
@goofmint 

  "Status": "DONE",
  "RequestHeaders": "{\"host\":\"125.1.77.160\",\"x-forwarded-for\":\"175.184.30.206, 175.184.41.116\",\"x-forwarded-host\":\"125.1.77.160\",\"x-forwarded-server\":\"125.1.77.160\",\"x-real-ip\":\"175.184.41.116\",\"connection\":\"close\",\"origin\":\"https://console.mb.cloud.nifty.com\",\"accept\":\"application/json, text/plain, */*\",\"user-agent\":\"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_1) AppleWebKit/604.3.5 (KHTML, like Gecko) Version/11.0.1 Safari/604.3.5\",\"referer\":\"https://console.mb.cloud.nifty.com/\",\"accept-language\":\"ja-jp\",\"accept-encoding\":\"br, gzip, deflate\"}",
  "RequestQuery": "{}",
  "ResponseHeaders": "{\"Content-Type\":\"application/json\"}",
  "Error": null,
  "Duration": 1266,
  "created_at": "2018-01-22T02:46:56.000Z",
  "updated_at": "2018-01-22T02:46:57.000Z",
  "ResponseStatus": 200,
  "ScriptIdentifier": "pushFlagDelete.js"
}

このようなログになっています！

**Comment by goofmint** [comment]
Response Bodyはこの内容からでは分かりませんが、空文字を返していたりするでしょうか？JSONを返すようにすると動くかな、と。

```js
res.send({})
```

だけでも。

**Comment by kiripeace** [comment]
@goofmint 
スクリプトにはそのように記入しています。
実際に動作もするのですが、実行時にGoogleAppScriptのほうからエラーが飛んでくるといった形になります。。

**Comment by goofmint** [comment]
そうですね。で、エラー行を見るにJSONのパースで失敗しているようなのです。なのでJSONではないもの（空文字など）が返ってきているのかなぁと思ったのですが。

**Comment by kiripeace** [comment]
空文字やJSONではないものはおそらく返してないと思うのですが、、、
もう少し調べたりコードを見直してみます、、
ご回答ありがとうございます！
